pyta_server/routes.py

```diff
@@ -1,32 +1,85 @@
 import os
 import secrets
 import psycopg2
+import json
 from flask import request
 from pyta_server import app, db
-from pyta_server.models import Uploads
+from pyta_server.models import Devices, Uploads, Files, Errors
+
+
+def update_db(entry):
+    """Save the entry to the database."""
+    try:
+        db.session.add(entry)
+        db.session.commit()
+    except (Exception, psycopg2.Error) as error:
+        print("Error while connecting to PostgreSQL", error)
+
+
+def commit_errors(error_info, upload, file=None):
+    """Make an Errors tuple and save it to the database."""
+    error = Errors(upload=upload,
+                   file=file,
+                   msg_id=error_info.get('msg_id'),
+                   msg=error_info.get('msg'),
+                   symbol=error_info.get('symbol'),
+                   category=error_info.get('category'),
+                   line=error_info.get('line'))
+    update_db(error)
+
 
 @app.route('/', methods=['POST'])
 def receive():
-    rand_hex = secrets.token_hex(8)
-    src_f = request.files.get('source')
-    cfg_f = request.files.get('config')
-    id = request.form.get('id')
-    if src_f is not None and cfg_f is not None:
-        _, src_ext = os.path.splitext(src_f.filename)
-        _, cfg_ext = os.path.splitext(cfg_f.filename)
-        src_n = rand_hex + src_ext
-        cfg_n = rand_hex + cfg_ext
-        src_path = os.path.join(app.root_path, 'static', 'source', src_n)
-        cfg_path = os.path.join(app.root_path, 'static', 'config', cfg_n)
-        src_f.save(src_path)
-        cfg_f.save(cfg_path)
-
-        upload = Uploads(source=src_path, config=cfg_path, identifier=id)
-        try:
-            db.session.add(upload)
-            db.session.commit()
-            return "files successfully received"
-        except (Exception, psycopg2.Error) as error:
-            print("Error while connecting to PostgreSQL", error)
+    """Receive POST request sent from user running PyTA."""
+    if isinstance(request.files, dict):
```

Comment:
This check is a bit awkward because it causes the (almost entire) rest of the function body to be indented. Can `request.files` be something other than a `dict`?

Even if so, check for this when extracting the source and config files from the request.